src/main/java/edu/cornell/kfs/module/cam/rest/resource/CuCapAssetInventoryApiResource.java

```diff
@@ -131,6 +146,45 @@ public Response updateAsset(@PathParam(CuCamsConstants.CapAssetApi.CAPITAL_ASSET
         }
     }
 
+    private void createCapitalAssetErrorDocument(String netid, String assetTag, String condition, String buildingCode, String roomNumber) {
+        try {
+            GlobalVariables.clear();
+
+            GlobalVariables.setUserSession(new UserSession("kp378"));
```

Comment:
Why is this using a hard coded net id for creating a user session? Are these documents supposed to be initiated by this user? Could you just set the initiator net id? Can this user created these documents manually as well? If so, can they differentiate between the ones automatically created by this integration and the ones they manually created?

Reply:
Setting the user session to kp378 allows the document initiator to be kp378 and it routes the document to kp378. I'm not sure how to set the document initiator to kp378 or a group?
I have found some base code that may be a clue as to how to do it in DocumentBase.java: wrapDocumentWithMetadataForXmlSerialization

Reply:
Using the UserSession will work, sorry for adding confusion on this one. I see that you have changed the initiator to be the kfs user. Will the routing work as expected in that case?  If it doesn't work then either the kfs user will need some role/permission added or if it is ok for kp378 to initiate these automatically created documents then perhaps we could use a system parameter to store this value. If one day this user no longer works for the university or she changes her job and another person should be responsible for these documents then a code change will be needed if we leave it hard coded. If we use a system parameter then the functional users can update the value in the application and this will keep working without needing a code change.

Reply:
There is a related SQL PR to add permission to kfs system user to create/initiate BCIE documents. I am currently working on routing to the capasset distribution group

Reply:
Thank you for pointing the SQL PR.
For the routing, will you need to change the workflow xml for the BCIE doc type?